src/test/java/com/microsoft/sqlserver/clientcertauth/ClientCertificateAuthenticationTest.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.clientcertauth;
+
+import static org.junit.Assert.assertTrue;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerDataSource;
+import com.microsoft.sqlserver.jdbc.SQLServerException;
+import com.microsoft.sqlserver.jdbc.TestResource;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.Constants;
+
+
+/**
+ * Tests client certificate authentication feature
+ * The feature is only supported against SQL Server Linux CU2 or higher.
+ * 
+ */
+@RunWith(JUnitPlatform.class)
+@Tag(Constants.xSQLv12)
+@Tag(Constants.xSQLv14)
+@Tag(Constants.xAzureSQLDW)
+@Tag(Constants.xAzureSQLDB)
+@Tag(Constants.clientCertAuth)
+public class ClientCertificateAuthenticationTest extends AbstractTest {
+
+    @Test
+    public void pkcs1Test() throws Exception {
+        String conStr = connectionString + ";clientCertificate=" + clientCertificate + ".pem;" + "clientKey="
+                + clientKey + "-pkcs1.key;";
+        try (Connection conn = DriverManager.getConnection(conStr); Statement stmt = conn.createStatement()) {
+            ResultSet rs = stmt.executeQuery("SELECT @@VERSION AS 'SQL Server Version'");
+            rs.next();
+            assertTrue(rs.getString(1).contains(TestResource.getResource("R_microsoft")));
+        }
+    }
+    
+    @Test
+    public void pkcs1EncryptedTest() throws Exception {
+        String conStr = connectionString + ";clientCertificate=" + clientCertificate + ".pem;" + "clientKey="
+                + clientKey + "-encrypted-pkcs1.key;" + "clientKeyPassword=" + clientKeyPassword + ";";

```

Comment:
We should not be hardcoding the file names like this. Please use config properties file and specify full file names in it.

Reply:
Why do we need to use a config file? The user can only provide one connection property for `clientCertificate`, and that one property needs to be re-used multiple times for different private key formats. The hardcoded strings here are the formats for the names that the test has defined. I can turn these hardcoded strings into constants, but I wouldn't use a config file here.